### ISSUE TITLE
no-unnecessary-callback-wrapper false positive

### ISSUE THREAD

**Issue by cartant** [palantir/tslint#2509]
### Bug Report

- __TSLint version__: 5.0.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: SublimeLinter

#### TypeScript code being linted

```ts
return this.patch(this.actions
    .ofType(...Object.keys(effects))
    .map((action) => effects[action.type](action))
    .first()
);
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "jsRules": {},
    "rules": {
        ...
        "no-unnecessary-callback-wrapper": { "severity": "warning" },
        ...
    }
}
```

#### Actual behavior

A warning is raised for the arrow function that's passed to the `map` operator.

#### Expected behavior

I would not have expected a warning, as the `action` parameter is used to look up the function that's to be called; it's not simply forwarded to a directly-callable function.
